addons/account/models/account_move.py

```diff
@@ -3230,6 +3223,47 @@ def _get_new_hash(self, secure_seq_number):
         #build and return the hash
         return self._compute_hash(prev_move.inalterable_hash if prev_move else u'')
 
+    def _hash_moves(self, force=False):
+        self.flush_recordset()  # Ensure that the name is correctly computed before it is used to generate the hash
+        for move in self._get_moves_to_hash(force=force):
+            new_number = move.journal_id.secure_sequence_id.next_by_id()
+            super(AccountMove, move).write({
+                'secure_sequence_number': new_number,
+                'inalterable_hash': move._get_new_hash(new_number),
+            })
+
+    def _get_moves_to_hash(self, force=False):
+        """
+        :return: the moves that need to be hashed taking into account all moves after the last
+                 hashed moved in the chain.
+        """
+        self = self.sorted(lambda m: (m.name, m.id))[-1:]
+        if not (
+            self
+            and
+            (
+                self._is_move_restricted(self) or self.move_type == 'entry' or force
```

Comment:
Why is there a case of `move_type` that is not included in `_is_move_restricted`?